hazelcast/src/main/java/com/hazelcast/internal/management/ManagementCenterService.java

```diff
@@ -46,36 +53,155 @@
  */
 public class ManagementCenterService {
 
+    static class MCEventStore {
+        static final long MC_EVENTS_WINDOW_MILLIS = TimeUnit.SECONDS.toMillis(30);
+        private final LongSupplier clock;
+        private volatile long mostRecentAccessTimestamp;
+        private final ConcurrentMap<Address, Long> lastAccessTimestamps = new ConcurrentHashMap<>();
+        private final ConcurrentMap<Address, Integer> eventsReceivedInSameMillisec = new ConcurrentHashMap<>();
+        private final BlockingQueue<Event> mcEvents;
+
+        MCEventStore(LongSupplier clock, BlockingQueue<Event> mcEvents) {
+            this.clock = clock;
+            this.mostRecentAccessTimestamp = clock.getAsLong();
+            this.mcEvents = mcEvents;
+        }
+
+        @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_BAD_PRACTICE")
+        void log(Event event) {
+            if (clock.getAsLong() - mostRecentAccessTimestamp > MC_EVENTS_WINDOW_MILLIS) {
+                // ignore event and clear the queue if the last poll happened a while ago
+                onMCEventWindowExceeded();
+            } else {
+                mcEvents.offer(event);
+            }
+        }
+
+        void onMCEventWindowExceeded() {
+            mcEvents.clear();
+            lastAccessTimestamps.clear();
+            eventsReceivedInSameMillisec.clear();
+        }
+
+        /**
+         * @param mcRemoteAddr the address of the calling MC instance.
+         * @return the events which were added to the queue since this MC last polled, or all known events if the MC polls for the
+         * first time.
+         */
+        public List<Event> pollMCEvents(Address mcRemoteAddr) {
+            Long lastAccessObj = lastAccessTimestamps.get(mcRemoteAddr);
+            mostRecentAccessTimestamp = clock.getAsLong();
+            long pollStartedAt = mostRecentAccessTimestamp;
+            List<Event> events;
+            if (lastAccessObj == null) {
+                System.out.println("null -> ");
```

Comment:
I guess this and following `System.out.println`s are debug leftovers and should be removed, right?